wolfcrypt/src/sha.c

```diff
@@ -36,16 +36,25 @@
 #ifdef HAVE_FIPS
 	int wc_InitSha(Sha* sha)
 	{
+        if (sha == NULL) {
+            return -173;
```

Comment:
Please address other areas in the code that return -173 directly as well.  They should return BAD_FUNC_ARG instead.